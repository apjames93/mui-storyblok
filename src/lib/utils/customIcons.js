const customIcons = [];


export const pushToCustomIcons = (iconArray) => {
  if (iconArray.length > 0) iconArray.forEach(icon => customIcons.push(icon));
};

export default customIcons;
