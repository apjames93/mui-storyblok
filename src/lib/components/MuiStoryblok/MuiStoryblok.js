import React from 'react';
import PropTypes from 'prop-types';
import GoogleApis from '../../utils/GoogleApis';
import StoryBlokPage from '../StoryBlokPage/StoryBlokPage';
import customIcons from '../../utils/customIcons';

export const MuiStoryblok = ({
  theme,
  accessToken,
  version,
  googleapisKey,
  useObjectTheme,
  iconArray,
}) => {
  // set GoogleApis instance to window to use in GeocodeTabs
  if (googleapisKey) window.muistoryblokgoogleapis = new GoogleApis(googleapisKey);
  if (iconArray.length > 0) iconArray.forEach(icon => customIcons.push(icon));

  return (
    <StoryBlokPage
      useObjectTheme={useObjectTheme}
      theme={theme}
      accessToken={accessToken}
      version={version}
    />
  );
};

export default MuiStoryblok;

MuiStoryblok.propTypes = {
  /** theme for mui */
  theme: PropTypes.shape(),
  /** storyblok prop  'published' : 'draft' */
  version: PropTypes.string,
  /** acess key from storyblok you can make them in storyblok settings */
  accessToken: PropTypes.string.isRequired,
  /** key for google apis Key to use geocode if here will make  */
  googleapisKey: PropTypes.string,
  /**
   * override url theme to edit in mui-theme tool
   */
  useObjectTheme: PropTypes.bool,
  /**
   * pass an array of icons to render custom icons in MuiIcon componet
   */
  iconArray: PropTypes.arrayOf(
    PropTypes.shape({
      iconName: PropTypes.string.isRequired,
      component: PropTypes.element.isRequired,
      props: PropTypes.shape({}).isRequired,
    }),
  ),
};

MuiStoryblok.defaultProps = {
  theme: {},
  version: 'draft',
  googleapisKey: '',
  useObjectTheme: false,
  iconArray: [],
};
