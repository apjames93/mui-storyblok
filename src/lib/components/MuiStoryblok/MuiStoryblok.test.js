import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiStoryblok from './MuiStoryblok';
import customIcons from '../../utils/customIcons';

function setup(iconArray = []) {
  const props = {
    accessToken: '123',
    googleapisKey: '1234',
    theme: { cool: 'value' },
    iconArray,
  };
  const comp = shallow(<MuiStoryblok {...props} />);
  return { comp, props };
}

describe('<MuiStoryblok />', () => {
  it('renders', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  it('iconArray', () => {
    const icons = [{
      iconName: 'thing',
      Component: <div>hey</div>,
    }];
    setup(icons);
    expect(customIcons).toEqual(icons);
  });


  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<MuiStoryblok {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
