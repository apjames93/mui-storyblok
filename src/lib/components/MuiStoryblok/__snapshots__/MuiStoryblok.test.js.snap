// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiStoryblok /> snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  style={Object {}}
>
  <div>
    Aloha
  </div>
  
</div>
`;
